controllers/webhook/runcompletion_feed.go

```diff
@@ -63,16 +70,32 @@ func getRequestBody(ctx context.Context, request *http.Request) ([]byte, error)
 	return body, nil
 }
 
-func extractEventData(ctx context.Context, request *http.Request) (*EventData, error) {
-	body, err := getRequestBody(ctx, request)
+type DataWrapper struct {
+	Data common.RunCompletionEventData `json:"data"`
+}
+
+func (rcf RunCompletionFeed) extractEventData(request *http.Request) (*EventData, error) {
+	body, err := getRequestBody(rcf.ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	runDataWrapper := &DataWrapper{}
+	if err := json.Unmarshal(body, runDataWrapper); err != nil {
+		return nil, err
+	}
+
+	rce, err := rcf.eventProcessor.ToRunCompletionEvent(rcf.ctx, runDataWrapper.Data)
 	if err != nil {
 		return nil, err
+	} else if rce == nil {
+		return nil, errors.New("event data is empty")
+	} else {
+		return &EventData{
+			Header:             request.Header,
+			RunCompletionEvent: *rce,
+		}, nil
 	}
```

Comment:
Minor but we typically split up if statements if they return. If you really prefer the if-elif-else then do that, up to you 👍 

```suggestion
	rce, err := rcf.eventProcessor.ToRunCompletionEvent(rcf.ctx, runDataWrapper.Data)
	if err != nil {
		return nil, err
	}
	
	if rce == nil {
		return nil, errors.New("event data is empty")
	}
	
	return &EventData{
		Header:             request.Header,
		RunCompletionEvent: *rce,
	}, nil
	}
```

Reply:
I think if ifelse else is better and less risk of introducing a bug in the execution flow